CRM/Utils/Array.php

```diff
@@ -148,7 +148,12 @@ public static function &xml(&$list, $depth = 1, $seperator = "\n") {
       else {
         // make sure we escape value
         $value = self::escapeXML($value);
-        $xml .= "<{$name}>$value</{$name}>{$seperator}";
+        if (is_numeric($name) && is_numeric($value) && ctype_digit(substr($name, 0, 1))) {
+          $xml .= "<id>$value</id>{$seperator}";
```

Comment:
The answer to (1) is "no" since the code checks the name not the value is numeric, so your example would still produce `<foo>5</foo>`. However I'm also concerned about the potential downstream impact of this change. Our rest api has been producing this output for a long time and no one has complained so far :/